### PR TITLE
Tests to verify correct (local) overwrite behavior

### PR DESCRIPTION
**Pull request by rrjjvv**
Verifies https://github.com/earthly/buildkit/pull/19, which in turn fixes https://github.com/earthly/earthly/issues/3094.

This ensures that an existing file is truncated prior to writing; otherwise, if the source (`X`) is smaller than the destination (`Y`), the destination contains all of `X` bytes at the beginning, and `Y-X` bytes of the original destination at the end.

### CODE REVIEW

**Comment by rrjjvv** [comment]
Hopefully I followed the correct procedure; let me know if I did anything wrong.  I assumed that I was *not* supposed to push the results of `+update-buildkit`.

Note that the test will fail without the buildkit changes:
```
./tests/local+test-local-overwrites *local* | WARN tests/local/Earthfile line 112:4: The command 'RUN set -e; for i in eight four two; do cat /tmp/earthly-test-21f37581-de77-4f81-ac78-4a2a617f7eda/${i}.txt | grep '^4321$'; done' failed: error calling LocalhostExec: exit code: 1
Error: build target: build main: failed to solve: tests/local/Earthfile line 112:4 apply RUN: unlazy force execution: error calling LocalhostExec: exit code: 1
in		github.com/rrjjvv/earthly/tests/local:local-copy-overwrite-fix+test-local-overwrites
```

Of course, they should pass when they're built against the changes in the other PR.

File: tests/local/Earthfile [hunk 1]
```diff
@@ -90,6 +90,30 @@ test-save-local-artifact-with-workdir:
...
     RUN cat data | grep '^ddd$'
 
+#################################################################################################################
+# tests for testing overwriting existing local files
+
+some-artifact:
+    RUN printf '4321' > four-bytes.txt
+    SAVE ARTIFACT four-bytes.txt
+
+test-local-overwrites:
```
Comment (alexcb) [inline tests/local/Earthfile:100]: can you add a `BUILD +test-local-overwrites` to the `all` target, which will then run this with our integration tests.

Comment (rrjjvv) [reply tests/local/Earthfile:100]: whoops, and done

**Comment by alexcb** [comment]
it's a bit of a multi-repo dance, but could you do a `./earthly +update-buildkit --BUILDKIT_GIT_SHA=cf348cf54d34478bb7c0ac62b0036802a6d59dd8` now that the buildkit change has been merged.

**Review by alexcb — APPROVED** [review]

**Pull request merged**